### ISSUE TITLE
Client update for Linux/Windows

### ISSUE THREAD

**Issue by madrang** [desura/Desurium#144]
I think it is time to plan for an update for Linux and Windows Desura client.
The latest update on Linux 64 is from Nov 26, soon it will be 6 month since the last update. It is now time to think about pushing an update.

**Comment by Jookia** [comment]
Have you seen the roadmap?

**Comment by madrang** [comment]
Yea... But there is no date. For all i know the next release could only be ready for next year.

**Comment by Jookia** [comment]
It seems wrong to set a date if we don't have a date.

**Comment by ghost** [comment]
I say that not having a date is better, simply because that doesn't rush the update just to make people happy.

**Comment by ssokolow** [comment]
I agree with @Jookia and @redhat69. Generally, release dates only provide more benefit than bother in two situations:
1. If you're undisciplined enough to need extra help avoiding feature creep. (In which case, explicit dates for feature freeze, string freeze, and release can help)
2. If inclusion in some other date-specific thing (like an Ubuntu release) depends on it.

Number 2 isn't an issue yet and, given how fresh the build system rewrite is, I don't think we have enough data to conclude that feature creep is a major issue.

**Comment by Jookia** [comment]
I am actually a tiny bit concerned about feature creep. The build system rewrite was necessary, but my work on #80 was suppose to be delayed until the mac client release, although it's better in the long run to do it now for compatibility's sake and distro packaging.

**Comment by ssokolow** [comment]
Perhaps there should be some sort of "milestone freeze" date then. That'd avoid the problems release dates tend to cause while still limiting the potential for feature creep. (You'd still have to stick to the milestones, which you didn't with #80, but it would probably be easier with smaller, more frequent releases. That's one of the secondary reasons Firefox went to their rapid release cycles.)

Of course, there'd still have to be a clear definition of what it means for a milestone to be frozen (For example, are newly-discovered crash bugs exempt?) and it may prove necessary to bump some issues to a different milestone if progress is slower than predicted.

**Comment by Jookia** [comment]
Setting this milestone for the next update, and tagging it as 'feature-big' and 'wishlist'.

**Comment by LiamDawe** [comment]
I think having a date for at least a feature freeze would be a good idea to then focus on polish for a while?

**Comment by Jookia** [comment]
If somebody wants to name a date (Lodle?), go ahead. Be sure to blame them if we fall through though.

**Comment by ssokolow** [comment]
How would one "fall through" a feature freeze when you're using something like Git with cheap branching?

**Comment by Jookia** [comment]
I mean the release date. Features are already unofficially frozen.

**Comment by Protektor-Desura** [comment]
For the next release I just want to see upload/download (#33) and Windows add/remove (#46) and validate games (#34) fixed and to me that fixes some major issues and it could be used as the next release. You for sure can't do a release until #33 is fixed that is a serious bug that must be fixed.

**Comment by LiamDawe** [comment]
I think rather than getting everything done that's on the list you should focus on the main bugs linked above, as the client from what i have seen is already miles ahead the current download. Then after that focus on other things, just to get something out, working and done.

**Comment by Protektor-Desura** [comment]
So have we agreed that once #33, #34 and #46 are fixed then we will do a release?

**Comment by Jookia** [comment]
So we'll do a quick release, then focus on what the milestone currently has?

**Comment by LiamDawe** [comment]
Move all the rest over to an Initial + 1? Milestone? Bar a few major needed ones like fixing settings (how do you quick link bugs like Protektor?)

**Comment by Jookia** [comment]
Screw that. Pre-initial?

**Comment by LiamDawe** [comment]
Well the name of the milestone is largely irrelevant. Issue 152 should probably be included too...

**Comment by Jookia** [comment]
I don't want to go through every single issue and push it to a new milestone. Oh wait, I could just rename the current one and move some to a new one.

**Comment by LiamDawe** [comment]
Yeah that would do it :D

**Comment by Protektor-Desura** [comment]
Yes we should fix #152 and get the settings menu working again.

**Comment by Jookia** [comment]
Okay, tell me if this sounds okay. A new milestone named 'FOSS Pre-release' that includes #33, #34, #46 and #152?

**Comment by LiamDawe** [comment]
Im happy with that.
